packages/utils/src/coverage-assessment.ts

```diff
@@ -0,0 +1,160 @@
+
+// query stuff
+const delayTime = dayjs.duration(3, "seconds");
+const numberOfParallelExecutions = 10;
+const confirmationTime = dayjs.duration(10, "seconds");
+
+// csv stuff
+const csvHeader =
+  "patientId,firstName,lastName,state,downloadStatus,docCount,convertStatus,fhirResourceCount,fhirResourceDetails\n";
+
+const csvName = (cxName: string): string =>
+  `./runs/coverage-assessment/${getFileNameForOrg(cxName, "csv")}`;
+
+const program = new Command();
+program
+  .name("coverage-assessment")
+  .description("CLI to get coverage/density data multiple patients.")
+  .showHelpAfterError();
+
+async function main() {
+  program.parse();
+  const { log } = out("");
+
+  const startedAt = Date.now();
+  log(`>>> Starting with ${patientIds.length} patient IDs...`);
+  const { orgName } = await getCxData(cxId, undefined, false);
+  const fileName = csvName(orgName);
+
+  initCsv(fileName);
+
+  const patientIdsToQuery = await getPatientIds(log);
+  log(`>>> Running it...`);
+
+  await executeAsynchronously(
+    patientIdsToQuery,
+    async (patientId, itemIdx) => {
+      await getCoverageForPatient(patientId, fileName, log);
+      log(`>>> Progress: ${itemIdx + 1}/${patientIdsToQuery.length} patients complete`);
+      await sleep(delayTime.asMilliseconds());
+    },
+    { numberOfParallelExecutions }
+  );
+
+  const ellapsed = Date.now() - startedAt;
+  log(`>>> Done assessing coverage for all ${patientIdsToQuery.length} patients in ${ellapsed} ms`);
+  process.exit(0);
+}
+
+function initCsv(fileName: string) {
+  const dirName = path.dirname(fileName);
+  if (!fs.existsSync(dirName)) {
+    fs.mkdirSync(dirName, { recursive: true });
+  }
+  fs.writeFileSync(fileName, csvHeader);
+}
+
+async function getPatientIds(log: typeof console.log): Promise<string[]> {
+  if (patientIds.length > 0) {
+    return patientIds;
+  }
+  return await getAllPatientIds(log);
+}
+
+async function getAllPatientIds(log: typeof console.log): Promise<string[]> {
+  displayNoDryRunWarning(log);
+  log(
+    "You are about to trigger a document query for all patients of the CX. This is very expensive!"
+  );
+  log("Cancel this script now if you're not sure.");
+  await sleep(confirmationTime.asMilliseconds());
+  const resp = await axios.get(`${apiUrl}/internal/patient/ids?cxId=${cxId}`);
+  const patientIds = resp.data.patientIds;
+  return (Array.isArray(patientIds) ? patientIds : []) as string[];
+}
+
+function displayNoDryRunWarning(log: typeof console.log) {
+  // The first chars there are to set color red on the terminal
+  // See: // https://stackoverflow.com/a/41407246/2099911
+  log("\n\x1b[31m%s\x1b[0m\n", "---- ATTENTION - THIS IS NOT A SIMULATED RUN ----");
+}
+
+async function getCoverageForPatient(patientId: string, fileName: string, log: typeof console.log) {
+  try {
+    const [patient, docQueryStatus, fhir] = await Promise.all([
+      sdk.getPatient(patientId),
+      getDocQueryStatus(cxId, patientId),
```

Comment:
I see the doc query status here but where we actually querying for patient docs?

Reply:
Or is the idea being that we ran that with the builk query and here were just checking it was actually made for the patient

Reply:
yeah, this is just to check the results 👍